src/main/java/org/opentripplanner/util/ProgressTracker.java

```diff
@@ -0,0 +1,47 @@
+package org.opentripplanner.util;
+
+import java.util.function.BiConsumer;
+
+
+/**
+ * The progress tracker notify the caller based on 'time' and a 'counter'.
+ * <p>
+ * The 'counter' is used to notify the caller for each percent(1%) it the total number of
+ * steps is more than 100. If the number steps is less than 100 the caller is notified every time.
+ * <p>
+ * To avoid the caller from being notified to often, the tracker also uses a 'timer'. The
+ * 'timer' prevent notification unless a minimum amount of time is passed since last time the
+ * caller was notified.
+ * <p>
+ * Both the 'counter' and the 'timer' constraint must pass for the caller to be notified.
+ */
+public class ProgressTracker {
+    private final int dStep;
+    private final int size;
+    private final long notifyEveryMillisecond;
```

Comment:
I would rename this to something like quietPeriodMilliseconds, because it doesn't ensure an update every M milliseconds, it silences updates for that long.